tests/jobs/test_backfill_job.py

```diff
@@ -637,8 +637,10 @@ def test_backfill_skip_active_scheduled_dagrun(self, dag_maker, caplog):
             end_date=DEFAULT_DATE + datetime.timedelta(days=2),
         )
         job.run()
-        error_log_records = [record for record in caplog.records if record.levelname == "ERROR"]
-        assert "Backfill cannot be created for DagRun" in error_log_records[0].msg
+        error_log_records = "\n".join(
+            [record.msg for record in caplog.records if record.levelname == "ERROR"]
+        )
+        assert "Backfill cannot be created for DagRun" in error_log_records
```

Comment:
I thought it also might happen because we do not filter by logger_name.
So instead of manual filter we could use context manager and just in case clear all captured logs before run job

```python
        with caplog.at_level(logging.ERROR, logger="airflow.jobs.backfill_job.BackfillJob"):
            caplog.clear()
            job.run()
            assert "Backfill cannot be created for DagRun" in caplog.messages[0]
```

Reply:
Indeed - should be better :). Added (and you as co-author :)